libraries/rush-lib/src/logic/operations/CacheableOperationPlugin.ts

```diff
@@ -658,20 +587,17 @@ export class CacheableOperationPlugin implements IPhasedCommandPlugin {
         return;
       }
 
-      const { outputFolderNames, stateHash: operationStateHash } = buildCacheContext;
+      const { outputFolderNames } = buildCacheContext;
       if (!outputFolderNames || !buildCacheConfiguration) {
         // Unreachable, since this will have set `cacheDisabledReason`.
         return;
       }
 
       // eslint-disable-next-line require-atomic-updates -- This is guaranteed to not be concurrent
-      buildCacheContext.operationBuildCache = ProjectBuildCache.getProjectBuildCache({
-        project: rushProject,
+      buildCacheContext.operationBuildCache = ProjectBuildCache.forOperation(operation, {
         projectOutputFolderNames: outputFolderNames,
```

Comment:
The only additional piece of information in `outputFolderNames` that isn't accessible off of `Operation` is the injection of the metadata folder; that could get moved inside of the build cache if it is being handed the entire `Operation` (or rather, the `OperationExecutionRecord`).